Show Changelog Feature

NunoLopesPT opened the pull request:
Pull Request for Issue # .

### Summary of Changes

Added a new parameter on the xml updateserver files 'changelogurl' and adding it to #__updates table.

### Testing Instructions

I made some changes in the core databases so before testing you should fix your database:

![screenshot from 2017-07-24 21-36-09](https://user-images.githubusercontent.com/20955884/28543880-43171938-70b9-11e7-9248-e3eff8b422d4.png)

Extensions -> Database -> Fix
Extensions -> Updates -> Find Updates

 **Update View** 

Install my component: https://github.com/NunoLopes96/component_joomla 
By default you will have always an update from 0.0.2 to 4.0.0 just for testing

If the changelog is not available it will show N/A like in the Folder column

![screenshot from 2017-05-23 18-28-40](https://cloud.githubusercontent.com/assets/20955884/26367191/ade139c4-3fe5-11e7-822f-345a2ab7ef34.png)


It will open a modal with an iframe of the changelogurl in the body.
The title of the modal is version ' - ' component_name

![screenshot from 2017-05-17 19-54-50](https://cloud.githubusercontent.com/assets/20955884/26170842/ba92eece-3b3a-11e7-8014-98ad32222650.png)

**Manage View**

The changelogurl is supported for Components, Libraries, Modules, Plugins and Packages, you just need to add <changelogurl> tags in your instalation xml with a valid url inside.

To test this you can use these modified packages:

(To test Package, Component, Module, Plugin)
[pkg_weblinks_3.4.1-changelog.zip](https://github.com/joomla-projects/gsoc17_expand_extension_manager/files/1127690/pkg_weblinks_3.4.1-changelog.zip)
![screenshot from 2017-07-06 12-34-02](https://user-images.githubusercontent.com/20955884/27909224-99574708-6247-11e7-9dc3-6071b909fd31.png)

(To test Libraries)
[lib_fof30-3.1.0-changelog.zip](https://github.com/joomla-projects/gsoc17_expand_extension_manager/files/1127689/lib_fof30-3.1.0-changelog.zip)
![screenshot from 2017-07-06 12-34-49](https://user-images.githubusercontent.com/20955884/27909264-cac5ff96-6247-11e7-8c23-936779591e82.png)

(To test languages)
The language installer its diferent from the others so you may want to test 2 cases:
1-Install without having it already in your extensions
2-Install without changelogurl and then install again with a changelogurl

This happens because it has 1 function to install and another one to update ( they are similar )

[admin_pt-PT.zip](https://github.com/joomla-projects/gsoc17_expand_extension_manager/files/1128171/admin_pt-PT.zip)
Extension name: Português (pt-PT)

(To test templates)
[et_kidtoy_joomla_template_install_package_free.zip](https://github.com/joomla-projects/gsoc17_expand_extension_manager/files/1128256/et_kidtoy_joomla_template_install_package_free.zip)
Extension name: etkidtoy

(To test Files)
[cli2zoombie.zip](https://github.com/joomla-projects/gsoc17_expand_extension_manager/files/1130860/cli2zoombie.zip)
Extension name: Zoombie - CLI Application

### Expected result

In Updates View you will have a new column that will show a changelog button to open a modal if there is a changelogurl in the #__updates table.
In Manage View you will have an <a> tag in the version with a info icon on the right side if your extension has a changelogurl in the #__extensions table.

### Documentation Changes Required

Add changelogurl parameter on updatesite information and in extension

zero-24 commented on administrator/components/com_installer/views/update/tmpl/default.php:
```diff
@@ -105,6 +105,15 @@
 									<td>
 										<span class="badge badge-success"><?php echo $item->version; ?></span>
+										<?php
+										$modal_params = array();
```
if you are in a layout file can you please use opening and closing PHP comands for each line?
So this line becomes
`<?php $modal_params = array(); ?>`

roland-d commented on administrator/components/com_installer/views/update/tmpl/default.php:
```diff
@@ -105,6 +103,14 @@
+
+										<a href="#myModal" class="btn" data-toggle="modal"> <?php echo $item->version; ?></a>
+
+										<?php echo JHTML::_('bootstrap.renderModal', 'myModal', $modal_params, $body); ?>
```
The class is spelled JHtml instead of JHTML

I think this is a nicer way of writing out the modal as you don't need to add the markup here.
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_categories/models/fields/modal/category.php#L188

alikon commented on administrator/components/com_installer/tmpl/update/default.php:
```diff
@@ -103,10 +104,33 @@
+										<?php
+										echo Jhtml::_(
+											'bootstrap.renderModal',
+											'changelog_modal',
```
bootstrap ???

alikon commented on administrator/components/com_admin/sql/updates/mysql/4.0.0-2017-06-03.sql:
```diff
@@ -0,0 +1 @@
+ALTER TABLE `#__extensions` ADD COLUMN `changelogurl` text AFTER `element`;
```
consider to merge in one file only

roland-d commented on administrator/components/com_installer/tmpl/manage/default.php:
```diff
@@ -106,7 +106,29 @@
 								</td>
 								<td class="hidden-sm-down text-center">
-									<?php echo @$item->version != '' ? $item->version : '&#160;'; ?>
+									<?php if($item->version != ''): ?>
```
You need a space after the if statement for codestyle. Check the rest of the code as well.

roland-d commented on administrator/components/com_installer/tmpl/update/default.php:
```diff
@@ -101,10 +102,33 @@
+
+										<?php else:?>
+										<span>
+											N/A
```
Shouldn't this be a JText string?

alikon commented:
testing the update view with these update sites
![updatesites](https://user-images.githubusercontent.com/181681/28560967-d81f341c-711c-11e7-9432-ed62a0fe1e95.png)

i've got this  cause that url doesn't not exists
![updatewarning](https://user-images.githubusercontent.com/181681/28561038-2316898e-711d-11e7-8eb4-6c8128d30d49.png)

and no other info are retrieved like from weblinks pakckage for example

NunoLopesPT commented:
@alikon I answered in Glip but in case you didn't saw the message, the website itself doesn't exist: http://www.joomladdons.eu/update/cli_zoombie_update.xml

alikon commented:
yes, but in my test weblinks is "updatable" and after the warning is not showed

NunoLopesPT commented:
I'm getting some trouble on getting separated extensions to test (When I did this pr they were working <.<).. Could we meet on Glip so I can help you testing this?

alikon commented:
me silly  :innocent:
the official weblink xml have ``` 
<targetplatform name="joomla" version="3.[6789]" />```
so wasn't updatable 
after hack targetplatform
![pr2ok](https://user-images.githubusercontent.com/181681/28709185-88b57624-737f-11e7-9131-b72eb8217604.png)

roland-d commented on administrator/components/com_installer/tmpl/update/default.php:
```diff
@@ -101,10 +102,33 @@
 									</td>
 									<td class="hidden-sm-down text-center">
+										<?php if ($item->changelogurl != null):?>
+										<button type="button" class="btn btn-info btn-xs" data-toggle="modal" data-target="#changelog_modal<?php echo $item->extension_id?>">Changelog</button>
```
The word Changelog should not be hardcoded but it needs to be a language string. There is a ; missing at the end of the last PHP code block

roland-d commented on libraries/src/CMS/Installer/Adapter/PackageAdapter.php:
```diff
@@ -548,6 +548,7 @@ protected function storeExtension()
 			$this->extension->name    = $this->name;
 			$this->extension->type    = 'package';
 			$this->extension->element = $this->element;
+			$this->extension->changelogurl = $this->changelogurl;
```
Align the variables.

roland-d commented on libraries/src/CMS/Installer/Adapter/TemplateAdapter.php:
```diff
@@ -520,8 +520,8 @@ protected function storeExtension()
 		{
-			$this->extension->type    = 'template';
-			$this->extension->element = $this->element;
+			$this->extension->type         = 'template';
```
The extra spaces are obsolete.

roland-d commented on libraries/src/CMS/Installer/InstallerAdapter.php:
```diff
@@ -671,6 +679,7 @@ public function install()
 		// Set the extension's name and element
 		$this->name    = $this->getName();
 		$this->element = $this->getElement();
+		$this->changelogurl = (string) $this->getManifest()->changelogurl;
```
Align the variables.

richard67 committed:
Remove template color handling from Cassiopeia offline.php (#23035)

PhilETaylor committed:
s/equivilents/equivalents & remove trailing periods from comments (#23005)

* s/equivilents/equivalents & remove trailing periods from comments

* s/equivilents/equivalents

PhilETaylor committed:
Remove commented out code. (#22991)

PhilETaylor committed:
[4.0] Install error pages typo and targe/rel on link to a user editable wikipage (#22996)

* s/Ses/Sets typo and rel/target

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

* build pages with npm

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
s/emtpy/empty (#23004)

PhilETaylor committed:
s/overriden/overridden (#22998)

* s/overriden/overridden

* Deliberately accept a spelling mistake in a 3rd party lib

reported upstream here https://github.com/jquery/jquery-ui/pull/1876

* revert using github online editor

* revert change and correct file ending as cannot be edited using github editor as that always adds a blank line!

* grrr

PhilETaylor committed:
s/tranistions/transitions & s/performecd/performecd (#23010)

s/tranistions/transitions & s/performecd/performecd

PhilETaylor committed:
s/interogate/interrogate (#23011)

PhilETaylor committed:
s/cummulative/cumulative/ (#23006)

PhilETaylor committed:
s/Nmber/Number (#23000)

PhilETaylor committed:
s/udpating/updating (#23002)

* s/udpating/updating

* s/statment/statement

PhilETaylor committed:
s/seprated/separated (#23003)

PhilETaylor committed:
s/presset/preset (#23012)

PhilETaylor committed:
s/informati/information & s/tha/that (#23013)

PhilETaylor committed:
s/resuls/results (#23014)

s/resuls/results

PhilETaylor committed:
s/belons/belongs (#23015)

s/belons/belongs

PhilETaylor committed:
s/protocals/protocols (#23016)

s/protocals/protocols

PhilETaylor committed:
s/seach/search (#23017)

s/seach/search

PhilETaylor committed:
s/informations/information (#23018)

s/informations/information

PhilETaylor committed:
s/postion/position (#23009)

PhilETaylor committed:
s/newfeed/newsfeed (#22999)

PhilETaylor committed:
s/buton/button (#23007)

SharkyKZ committed:
Return parent method value (#23048)

PhilETaylor committed:
Fix JS alerts for Joomla/Extension update check (#22985)

* Fix JS alerts for Joomla/Extension update check

Closes #22971
Related #22981

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

* codestyle

* Codestyle

* Improve documentation of function

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

* Add message type `warning` and downgrade message from error to a warning

* add warning example to docblock

laoneo committed:
Remove double empty lines in joomlaupdate JS

PhilETaylor committed:
s/succesfully/successfully (#23008)

wilsonge committed:
Fixes creating a new banner (thanks @phalouvas )

richard67 committed:
Remove module class _menu from menu modules parameters (#23054)

Remove module class _menu from menu modules parameters in new installation and sample data

Commit:
Fix markup (#23080)

SharkyKZ committed:
[4.0][com_workflows] Remove unused function (#23027)

yvesh committed:
[4.0][Testing] Updating drone.yml to always update JoomlaBrowser (#23094)

Updating JoomlaBrowser to the newest version for every test run.

laoneo committed:
Fix the batch layout identifiers (#23092)

Fedik committed:
[4.0] Fix undefined Awesomplete (#23113)

laoneo committed:
[4.0] Check if rules are available in rules field rule (#23125)

* Check if rules are available in rules field rule

* Update libraries/src/Form/Rule/RulesRule.php

Co-Authored-By: laoneo <allon.moritz@digital-peak.com>

infograf768 committed:
[4.0] System Panel: Add badges to items when necessary (#23047)

SharkyKZ committed:
[4.0] Fix retrieval of guest user group (#23085)

* [4.0] Fix retrieval of guest user group

* Cleanup

Commit:
Update IFW to work with CE, remove JSONP requirement (#23104)

wilsonge committed:
Fix styling of cancel button in com_content batch. Fixes #22886 (#23137)

wilsonge committed:
Upgrade custom elements to version 0.0.30 (#23145)

infograf768 committed:
[4.0] RTL: Correcting Joomla Update version display in System CPanel badge and Joomla Update component (#23140)

PhilETaylor committed:
[4.0] Add missing eye icons to installer css (#23097)

infograf768 committed:
[4.0] RFC Quickicons Joomla and Extensions updates (#23095)

infograf768 committed:
[4.0] Can't create field group (#23187)

infograf768 committed:
[4.0] Implementing RTL for admin menus (#23143)

infograf768 committed:
[4.0] RTL: Correcting broken installation (#23208)

laoneo committed:
Cleanup the imports (#23173)

SharkyKZ committed:
[4.0] Correct Table::hasField() uses (#23139)

* [4.0] Fix Table::hasField() uses

* CS

* Update libraries/src/MVC/Model/FormModel.php

Co-Authored-By: SharkyKZ <sharkykz@gmail.com>

* Update FormModel.php

* Update libraries/src/Table/Table.php

Co-Authored-By: SharkyKZ <sharkykz@gmail.com>

* Update libraries/src/Table/Table.php

Co-Authored-By: SharkyKZ <sharkykz@gmail.com>

SharkyKZ committed:
Typo (#23220)

infograf768 committed:
[4.0] Joomla update: correcting missing code from #23095 (#23214)

wilsonge committed:
[4.0] Add classes to represent help screens and move to the model (#23196)

infograf768 committed:
[4.0] RTL admin various corrections (#23230)

Commit:
Fix for Factory reference in JsonDocument ref #21304 (#23234)

infograf768 committed:
[4.0] Multilingual: correcting deprecated method in language filter (#23231)

dgrammatiko committed:
[4.0] npm build tools update (#23158)

wilsonge committed:
Fix several TODOs in code (#23238)

SharkyKZ committed:
[4.0] Fix notice in layouts/joomla/form/field/combo.php (#23120)

HLeithner committed:
Use break instead of continue in switch statement. (#23253)

Fedik committed:
[4.0] WebAsset for  JHtmlJquery,  JHtmlSearchtools (#23112)

dgrammatiko committed:
[4.0] Custom Elements fix - editor none (#23169)

dgrammatiko committed:
[4.0] Align editors to the rest of the plugins  (#23167)

bembelimen committed:
[4.0] Protect the default workflow (#22827)

bembelimen committed:
[4.0] Workflow prevent trashing of used stages/workflows (#21579)

bembelimen committed:
Respect stage ordering (#21705)

Hackwar committed:
[4.0] Smart Search: Adding debug information (#22593)

wojsmol committed:
Fixing composr.lock file (#23271)

wilsonge committed:
Merge commit '9780ed3' into 4.0-dev

wilsonge committed:
Merge branch '4.0-dev' into 4.0-dev

wilsonge committed:
Merge pull request #23280 from wilsonge/4.0-dev

Merge in part of Joomla 3.8.8

infograf768 committed:
[4.0] Translating Joomla default workflow title (#23275)

laoneo committed:
[4.0] Revert to jQuery in menus module selection till CE modals are used (#23062)

richard67 committed:
Remove menu class nav-pills (#23055)

laoneo committed:
Update Joomla browser with system test fixes (#23093)

wilsonge committed:
Merge commit 'f632960' into 4.0-dev

wilsonge committed:
Merge commit 'adb7651' into 4.0-dev

wilsonge committed:
Merge commit 'ff7addc' into 4.0-dev

wilsonge committed:
Merge commit 'b950107' into 4.0-dev

wilsonge committed:
Merge commit '4077a74' into 4.0-dev

wilsonge committed:
[CLEAR CACHE] Fix codestyle

wilsonge committed:
Merge pull request #23285 from wilsonge/4.0-dev

Merge in most the rest of Joomla 3.8.8

chmst committed:
[4.0] Improve a11y for systeminformation (#21724)

Commit:
Fix markup (#23299)

elkuku committed:
Fix missing line numbers in language file errors when xdebug.file_link_format is not set (#23308)

chmst committed:
[4.0][a11y] Sysinfo table caption (#23314)

ReLater committed:
correct typo unkown instead of unknown (#23313)

chmst committed:
[4.0] [a11y] Views user/ usergroup permissions: Improve accessibility (#23332)

bembelimen committed:
[4.0] Workflow article buttons (#22393)

chmst committed:
The dismiss button is a button, not a link (#23340)

wilsonge committed:
Merge tag '3.8.8' into 4.0-dev

3.8.8 Release3.8.8 Release3.8.8 Release3.8.8 Release3.8.8 Release3.8.8 Release3.8.8 Release3.8.8 Release

wilsonge committed:
Fix codestyle

wilsonge committed:
Fix error in installation

wilsonge committed:
Fix editing Access Levels

zero-24 committed:
Merge branch '4.0-dev' into 388-merge2

wilsonge committed:
Merge pull request #23343 from wilsonge/388-merge2

Merge in the rest of 3.8.8

bembelimen committed:
[4.0] Group toolbar buttons for com_menu (#23336)

* Group action buttons for com_menu

* Update buttons to new trait methods

bembelimen committed:
Group action buttons for com_categories (#23337)

wilsonge committed:
[4.0] Update database package to fix menu manager (#23344)

Hackwar committed:
Removing obsolete code from ComponentHelper (#23353)

wilsonge committed:
[4.0] Fixing bugs in menu manager (#23355)

wilsonge committed:
Tag Alpha 6

wilsonge committed:
Set back to development

wilsonge committed:
Merge commit '56b89ce' into 4.0-dev

wilsonge committed:
Merge commit '66debe9' into 4.0-dev

wilsonge committed:
Merge commit '8718160' into 4.0-dev

wilsonge committed:
Merge commit 'f35af12' into 4.0-dev

wilsonge committed:
Merge commit 'cfa9bcef88f240d030d9e22e2fd33a2b63ebbdac' into 4.0-dev

wilsonge committed:
Fix codestyle

wilsonge committed:
Merge pull request #23373 from wilsonge/389-merge

Start merging 3.8.9 into J4

wilsonge committed:
Merge commit '8afc860' into 4.0-dev

wilsonge committed:
Fix codestyle

wilsonge committed:
Merge pull request #23379 from wilsonge/389merge-2

3.8.9 Merge part 2

wilsonge committed:
Fix javascript lint (#23381)

dgrammatiko committed:
[4.0] ESLint javascript scripts or fail (#23363)

wilsonge committed:
Merge commit '6770b31' into 4.0-dev

wilsonge committed:
Merge pull request #23387 from wilsonge/389-merge3

3.8.9 Merge part 3

wilsonge committed:
Merge commit '7a8795d' into 4.0-dev

wilsonge committed:
Merge commit '8383f60' into 4.0-dev

wilsonge committed:
Merge pull request #23389 from wilsonge/389-merge4

Merge in 3.8.9 and 3.8.10

bembelimen committed:
Remove trash button from article toolbar (#21532)

dgrammatiko committed:
[4.0] Drop for loops (#23362)

zero-24 committed:
make sure the SECURITY.md is not included in the final packages (#23411)

dgrammatiko committed:
[4.0] Eliminate jQuery from Modal footer (#23413)

dgrammatiko committed:
[4.0] No jQuery in the xml files... (#23418)

csthomas committed:
[4.0] Use DebugMonitor from joomla/database vendor (#23375)

zero-24 committed:
CSP vs. Content Security Policy (#23412)

mbabker committed:
Fix creating metadata record for guest user

mbabker committed:
Wire the session MetadataManager as a service and listen to the session start event to trigger it

mbabker committed:
Change to create or update

infograf768 committed:
[4.0] Extensions Update Message: correcting #23362 (#23434)

chmst committed:
Fix notice in newsfeed (#23436)

Hackwar committed:
[4.0] Adding a node object interface (#23364)

* Adding a node object interface

* Making some methods nullable

* Adding getRoot

infograf768 committed:
[4.0] Admin modules: Cannot select or filter by position (#23289)

wilsonge committed:
Add an interface for views in Joomla (#23431)

bembelimen committed:
Remove unused postSaveHook methods (#23445)

dgrammatiko committed:
[4.0] No jquery, custom fields (#23414)

dgrammatiko committed:
[4.0] no jquery workflow button (#23417)

dgrammatiko committed:
[4.0] No jQuery (#23416)

infograf768 committed:
[4.0] Reimplementing client filter in Modules Manager (#23406)

bembelimen committed:
Hide "All" option in stage edit (#23443)

joomdonation committed:
Remove un-used line of code (#23394)

bembelimen committed:
Fix user notes state change (publish/unpublish/trash) (#23384)

bembelimen committed:
[4.0] Cleanup content/banner toolbar code (#23369)

chmst committed:
Group toolbar buttons for newsfeeds (#23404)

Co-Authored-By: chmst <dev@chmst.de>

chmst committed:
Group toolbar buttons in com_plugins into a status change dropdown (#23391)

Hackwar committed:
[4.0] Fixing lots of issues in Tags component and grouping toolbar buttons (#23402)

bembelimen committed:
[4.0] Group contact toolbar buttons (#23367)

Co-Authored-By: bembelimen <bembelimen@users.noreply.github.com>

bembelimen committed:
Fix wrong status of custom field enable switcher in com_contact (#23339)

chmst committed:
[4.0] Group action buttons in users (#23374)

bembelimen committed:
[4.0] Group action buttons for com_fields (#23370)

bembelimen committed:
Group action buttons for com_fields group (#23371)

chmst committed:
[4.0] Dispatcher access check (#23467)

chmst committed:
[4.0] Group action buttons for banner clients (#23368)

bembelimen committed:
[4.0] Group action buttons for com_modules (#23378)

infograf768 committed:
[4.0] JFactory error in PLuginsField.php (#23471)

infograf768 committed:
[4.0] correcting/adding JFactory to Factory (pass 1) (#23475)

infograf768 committed:
[4.0] Modules Manager: normalize client filter code (#23472)

bembelimen committed:
Fix features view and add grouped buttons (#23459)

chmst committed:
[4.0] Group action toolbar buttons usernotes (#23380)

chmst committed:
[4.0] Group toolbar buttons for links (#23390)

wilsonge committed:
Fix code style (#23477)

laoneo committed:
Use the namespaced Form class in the fields plugin functions (#22865)

bembelimen committed:
[4.0] Add back button to the transitions and stages view button toolbars (#23460)

laoneo committed:
[4.0] Introduce model interfaces (#20640)

Fedik committed:
[4.0] WebAsset: split WebAssetRegistry to Registry and Manager (#23463)

SharkyKZ committed:
[4.0] Missing files in template manifests (#23484)

wilsonge committed:
Merge commit '8ac66ea' into 4.0-dev

wilsonge committed:
Merge commit '899e7c9' into 4.0-dev

wilsonge committed:
Fix codestyle

wilsonge committed:
Merge pull request #23494 from wilsonge/3811-merge

3.8.11 merge work

chmst committed:
fix notice in syndicate.xml (#23502)

Hackwar committed:
[4.0] Not only set the redirect but actually return (#23498)

Denitz committed:
Undefined class \Joomla\CMS\Filesystem\AKFactory (#23506)

Fix `Undefined class \Joomla\CMS\Filesystem\AKFactory` on core files reinstall from https://update.joomla.org/core/nightlies/next_major_list.xml

Commit:
Update draggable.js (#23487)

chmst committed:
Finder Index: Check access to plugins (#23509)

wilsonge committed:
Reverse order of template params for green/red. Fixes #23519 (#23521)

Scrabble96 committed:
Update en-GB.com_templates.ini (#23523)

The existing instructions are for Joomla 3x

wilsonge committed:
[4.0] Fix Legacy Model loading. Fixes #23517 (#23525)

Co-Authored-By: wilsonge <georgejameswilson@googlemail.com>

ReLater committed:
Add $data to event_before_save in tags (#23531)

infograf768 committed:
[4.0] [com_menus] Fix "Too few arguments to function (#23535)

PlgContentJoomla::onContentBeforeSave"

ReLater committed:
[4.0] Add json to source_formats of template editor (#23470)

joomdonation committed:
[4.0] Remove show_associations field of association is not enabled (#23501)

dgrammatiko committed:
Move admin menu JS code to es6 (#23155)

bembelimen committed:
Fix missing use statement (#23564)

HLeithner committed:
Csp doc block change. (#23579)

Hackwar committed:
[4.0] Adding filter() and postProcessing() to FormField (#12414)

hardik-codes committed:
Fixed inconsistent use of plurals Ref #22791 (#23515)

* Update en-GB.mod_menu.ini

* Update en-GB.mod_quickicon.ini

Hackwar committed:
Smart Search searching performance improvement (#23495)

ReLater committed:
Remove folder "img" that doesn't exist from atum (#23594)

chmst committed:
Group toolbar buttons for workflows (#23605)

brianteeman committed:
[4.0] Typo in js comments (#23676)

Simple PR to correct a typo

brianteeman committed:
[4.0]Sample Data Module xml description to better English (#23681)

mbabker committed:
Remove myself from most CODEOWNERS paths (#23707)

The stats plugin is retained as this integrates with the .org architecture, which I remain engaged with

This commit is a cherry-pick of 0c83fe0ee42 from staging to 4.0-dev

chmst committed:
[4.0][a11y]Admin module logged button (#23277)

* Replace the logout link by a button

* Changes as required, quoting in querry

* revert quoting, remove space

* remove span

* Revert changes in helper uls logout link in form

* Update administrator/modules/mod_logged/tmpl/default.php

Co-Authored-By: chmst <dev@chmst.de>

* Remove Route::

* Remove icon

Hackwar committed:
com_finder categories plugin needs to check right extension element (#20658)

bembelimen committed:
Set inital workflow for the uncategories category to "use default" (#23701)

Hackwar committed:
[4.0] com_finder: Filtering common & numeric words and language fixes and params (#21037)

dgrammatiko committed:
[4.0] Housekeeping (#23600)

infograf768 committed:
[4.0] Correcting icons display in debug bar (#23732)

infograf768 committed:
[4.0] Correcting success colour after installing sample data (#23726)

infograf768 committed:
[4.0] Normalize sample data plugins to use Joomla default workflow (#23719)

SniperSister committed:
[4.0] Fix database name validation for MySQL in installation app (#23704)

infograf768 committed:
[4.0] Fix save contact (#23688)

Commit:
Remove libraries/cms/html/language/en-GB/en-GB.jhtmldate.ini (#23653)

laoneo committed:
Clean the node modules in the media folder on install (#22860)

laoneo committed:
[4.0] Remove deprecated functions isSite and isAdmin (#22492)

C-Lodder committed:
Fix stats alert information (#23727)

brianteeman committed:
[4.0] Remove unused string (#23750)

I assume that this string was intended to be used for the dropdown menu that is in the top right hand corner. IIRC it was at one point in atum's life hard coded into the template. However today that user menu is controlled by the mod_status administrator module and the language string used is from there and not from the template

Commit:
Remove extra type attribute (#23753)

chmst committed:
[4.0] [a11y] Make ActionButton Featured accessible (#23718)

dgrammatiko committed:
Fixes plugins ordering (#23731)

dgrammatiko committed:
[4.0] JS error when closing a item and going back to a list view (#23642)

alikon committed:
[4.0] - fix save article (#23638)

bembelimen committed:
Clean up hard coded usage of state values (#23330)

amitranjan2 committed:
[4.0] FIX SQL int default value to  numeric (#23669)

alikon committed:
modified_time check (#23689)

bembelimen committed:
[4.0] Set default workflow for protected workflows (#23607)

Co-Authored-By: bembelimen <bembelimen@users.noreply.github.com>

dgrammatiko committed:
Remove the slider button (#23388)

infograf768 committed:
[4.0] remove duplicate string (#23771)

wilsonge committed:
Fix saving content menu items (#23716)

dgrammatiko committed:
[4.0][NO CACHE] Refactor the build tools (#23723)

wilsonge committed:
Fix running npm install and npm ci (#23788)

wilsonge committed:
Some webcomponents already pass linting (#23789)

brianteeman committed:
[4.0] Disable Back to Control Panel link (#23804)

In Joomla 3.x we disabled the control panel link in the top left hand corner of the admin at the same time that we disabled the main menu. Amongst other reasons this was to prevent people from leaving an item checked out. This PR fixes that.

Co-Authored-By: brianteeman <brian@teeman.net>

dgrammatiko committed:
[4.0] fix media edit modal cancel button (#23880)

* Fix 23832

* Fix also the reset

* Update edit-images.es6.js

* Update edit-images.es6.js

bahl24 committed:
[4.0]fixed language of 'insert image' to 'Insert Image' (#23878)

* fixed language
* defined string in lang file

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

brianteeman committed:
[4.0] Plugins Change status (#23797)

Correction to #23991

Plugins are enabled/disabled NOT published/un published

wojsmol committed:
[4.0][a11n] aria-labeledby to aria-labelledby (#23834)

Hackwar committed:
[4.0] Making CategoryNode and MenuItem ImmutableNodes (#23452)

brianteeman committed:
[4.0][a11y] Labels Module Assignment are not associated with forms (#23922)

brianteeman committed:
[4.0] User Notes Category tags (#23921)

Remove class from tags field to prevent extra line being inserted

PR for #23920

brianteeman committed:
[4.0] Field Display when read only (#23930)

astridx committed:
[4.0] Improve a Language String (#23855)

hardik-codes committed:
[4.0] Fix calendar tweaks (#23903)

C-Lodder committed:
[4.0] Fancy Select custom element improvements (#23768)

Co-Authored-By: C-Lodder <lodder15@hotmail.co.uk>

ciar4n committed:
[4.0] Style legend the same as label (#23927)

chmst committed:
Fix aria attribute aria-sort (#23662)

mbabker committed:
Update joomla/console package and adapt code to changes (#23847)

dneukirchen committed:
Update browserify (#23953)

drmenzelit committed:
Changes in Toolbar, separate Save (and stay) button from Save button … (#23942)

brianteeman committed:
[4.0] Checkboxes [a11y] (#23970)

bahl24 committed:
Added placeholder for alias field in tags (#23940)

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

bembelimen committed:
System dashboard (#23936)

Co-Authored-By: bembelimen <bembelimen@users.noreply.github.com>

bembelimen committed:
[4.0] Workflow new article stage category change (#23458)

dgrammatiko committed:
[4.0] Do NOT force jQuery.js and Bootstrap.js in the backend template (#23742)

C-Lodder committed:
[4.0] Permissions custom element improvements (#23729)

C-Lodder committed:
[4.0] Hidden-mail custom element improvements (#23730)

brianteeman committed:
[4.0] Wrong string in com_users (#23901)

The wrong string (that was missing) was being used.

NOTE Any a11y issue regarding this is beyond the scope of this PR

bahl24 committed:
[4.0] Fixes issue #23746- unclear sidebar (#23747)

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

alikon committed:
[com_tag] - class not found (#23990)

roland-d committed:
Merge branch '4.0-dev' into show_changelog

# Conflicts:
#	README.md
#	administrator/components/com_installer/tmpl/manage/default.php
#	administrator/components/com_installer/tmpl/update/default.php
#	libraries/src/CMS/Installer/Adapter/ComponentAdapter.php
#	libraries/src/CMS/Installer/Adapter/FileAdapter.php
#	libraries/src/CMS/Installer/Adapter/LanguageAdapter.php
#	libraries/src/CMS/Installer/Adapter/LibraryAdapter.php
#	libraries/src/CMS/Installer/Adapter/ModuleAdapter.php
#	libraries/src/CMS/Installer/Adapter/PackageAdapter.php
#	libraries/src/CMS/Installer/Adapter/PluginAdapter.php
#	libraries/src/CMS/Installer/Adapter/TemplateAdapter.php
#	libraries/src/CMS/Installer/InstallerAdapter.php
#	libraries/src/CMS/Updater/UpdateAdapter.php

roland-d committed:
Propagate changes to new files

Signed-off-by: Roland Dalmulder <contact@rolandd.com>
